h2o-core/src/main/java/hex/CFuncScoringTask.java

```diff
@@ -0,0 +1,63 @@
+package hex;
+
+import water.udf.CFuncRef;
+import water.udf.CFuncTask;
+import water.udf.CMetricFunc;
+
+public class CFuncScoringTask<T extends CFuncScoringTask<T>> extends CFuncTask<CMetricFunc, T> {
```

Comment:
This should be renamed to reflect that this class handles custom metrics. Too generic now.